### PR TITLE
EDF random forest classifier samples

### PR DESCRIPTION
**Pull request by macnb9**
### What
- Improved the sample for training a classifier on a single patient EDF
- Added new example for batch training with multiple patient samples

### CODE REVIEW

File: mne-demo/plot_sleep_batch.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+annotation_desc_2_event_id = {'Sleep stage W': 1,
+                              'Sleep stage 1': 2,
+                              'Sleep stage 2': 3,
+                              'Sleep stage 3': 4,
+                              'Sleep stage 4': 4,
+                              'Sleep stage R': 5}
+
+all_events = []
+for raw_sample in all_raw_edf:
+    events, _ = mne.events_from_annotations(raw_sample, event_id=annotation_desc_2_event_id, chunk_duration=30.)
```
Comment (dkapp7) [inline mne-demo/plot_sleep_batch.py]: I see 30 is being used often. Have you tried different values for this? Would we want to make it a variable for easy tweaking?

Comment (macnb9) [reply mne-demo/plot_sleep_batch.py]: Ah yeah, I can go ahead and make these functions a bit more modular, I'll ping you when I push the changes

**Comment by macnb9** [comment]
@dkapp7 updated the code to be more modular. The one thing about the chunk_duration is that, even though we can modify that value in this dataset sample due to the sleep stages being a channel inside the edf file itself, the SHHS data is locked in with stage events predefined at exactly 30 second intervals, so we'll be stuck with that moving forward with the actual data.

File: mne-demo/plot_sleep_batch.py (hunk 1 above)
Comment (rongncao) [reply mne-demo/plot_sleep_batch.py]: I think it's because each event is 30 sec long

**Review by rongncao — APPROVED** [review]

**Pull request merged**